lib/oli/delivery/evaluation/rule.ex

```diff
@@ -142,10 +142,23 @@ defmodule Oli.Delivery.Evaluation.Rule do
   # checks the precision
   defp check_precision(str, count) when is_binary(str) do
     digit_count =
-      str
-      |> String.split("")
-      |> Enum.filter(&is_digit?/1)
-      |> Enum.count()
+      case String.split(str, ".") do
+        ["0", decimal] ->
+          decimal
+          |> String.replace_leading("0", "")
+          |> String.split("")
+          |> Enum.filter(&is_digit?/1)
+          |> Enum.count()
```

Comment:
Nit but these functions are called in the three cases, so what do you think about moving them to a separate function? It'd result in something like:
```elixir
defp count_digits(string_number) do
  string_number
  |> String.split("")
  |> Enum.filter(&is_digit?/1)
  |> Enum.count()
end

...

case String.split(str, ".") do
  ["0", decimal] ->
    decimal
    |> String.replace_leading("0", "")
    |> count_digits()

  [_integer, _decimal] ->
    count_digits(str)

  [integer] ->
    count_digits(integer)
end
```